tests/sentry/workflow_engine/endpoints/test_organization_detector_workflow_index.py

```diff
@@ -0,0 +1,220 @@
+            status_code=200,
+        )
+        assert len(response.data) == 2
+
+        response = self.get_success_response(
+            self.organization.slug,
+            qs_params={"detector_id": self.unconnected_detector.id},
+            status_code=200,
+        )
+        assert len(response.data) == 0
+
+    def test_workflow_filter(self):
+        response = self.get_success_response(
+            self.organization.slug,
+            qs_params={"workflow_id": self.workflow_1.id},
+            status_code=200,
+        )
+        assert len(response.data) == 2
+        assert response.data == [
+            {
+                "id": str(self.detector_1_workflow_1.id),
+                "detectorId": str(self.detector_1.id),
+                "workflowId": str(self.workflow_1.id),
+            },
+            {
+                "id": str(self.detector_2_workflow_1.id),
+                "detectorId": str(self.detector_2.id),
+                "workflowId": str(self.workflow_1.id),
+            },
+        ]
+
+        response = self.get_success_response(
+            self.organization.slug,
+            qs_params={"workflow_id": self.unconnected_workflow.id},
+            status_code=200,
+        )
+        assert len(response.data) == 0
+
+    def test_detector_workflow_filter(self):
+        response = self.get_success_response(
+            self.organization.slug,
+            qs_params={"detector_id": self.detector_1.id, "workflow_id": self.workflow_1.id},
+            status_code=200,
+        )
+        assert len(response.data) == 1
+        assert response.data == [
+            {
+                "id": str(self.detector_1_workflow_1.id),
+                "detectorId": str(self.detector_1.id),
+                "workflowId": str(self.workflow_1.id),
+            }
+        ]
+
+        response = self.get_success_response(
+            self.organization.slug,
+            qs_params={"detector_id": self.detector_2.id, "workflow_id": self.workflow_2.id},
+            status_code=200,
+        )
+        assert len(response.data) == 0
+
+
+@region_silo_test
+class OrganizationDetectorWorkflowIndexPostTest(OrganizationDetectorWorkflowAPITestCase):
+    method = "post"
+
+    def test_simple(self):
+        body_params = {
+            "detectorId": self.unconnected_detector.id,
+            "workflowId": self.unconnected_workflow.id,
+        }
+        response = self.get_success_response(
+            self.organization.slug,
+            **body_params,
+            status_code=200,
+        )
+        detector_workflow = DetectorWorkflow.objects.get(
+            detector_id=self.unconnected_detector.id, workflow_id=self.unconnected_workflow.id
+        )
+        assert response.data == {
+            "id": str(detector_workflow.id),
+            "detectorId": str(self.unconnected_detector.id),
+            "workflowId": str(self.unconnected_workflow.id),
+        }
+
+    def test_audit_entry(self):
```

Comment:
Here's an alternate way to add in checking audit log entries. Mock it like this: https://github.com/getsentry/sentry/blob/master/tests/sentry/workflow_engine/endpoints/test_project_detector_index.py#L121 and then check it here: https://github.com/getsentry/sentry/blob/master/tests/sentry/workflow_engine/endpoints/test_project_detector_index.py#L168-L175

Reply:
hmm i just tried this but i don't see a way to check for multiple audit log entries this way? i need to assert that all DetectorWorkflows have an audit log in the case of bulk deletion

Reply:
There are several options for mock assertion (docs [here](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.assert_called)) so you don't just need to assert it's called once with, maybe you could loop over the `DetectorWorkflow` objects and `assert_called_with`

Reply:
i realize that i messed something up the first time, i'll try again ty 🫡